src/traverse.js

```diff
@@ -0,0 +1,34 @@
+import { base58btc } from 'multiformats/bases/base58'
+
+/**
+ * @typedef {import('./cid.js').CID} CID
+ */
+
+/**
+ * @template T
+ * @typedef {import('./block.js').Block<T>} Block
+ */
+
+/**
+ * @template T
+ * @param {Object} options
+ * @param {CID} options.cid
+ * @param {(cid: CID) => Promise<Block<T>>} options.load
+ * @param {Set<string>} options.seen
```

Comment:
I think we need to mark this one as optional, `Set<string>?`

Reply:
Sure - I have added this too